### ISSUE TITLE
Deploying a server returns a 404 for an existing keypair

### ISSUE THREAD

**Issue by alipievg** [stackitcloud/terraform-provider-stackit#580]
**Describe the bug**

When using the resource stackit_server, terraform fails to deploy a server with a response:
```
╷
│ Error: Error creating server
│
│   with stackit_server.test,
│   on vms.tf line 1, in resource "stackit_server" "test":
│    1: resource "stackit_server" "test" {
│
│ Calling API: 404 Not Found, status code 404, Body: {"code":404,"msg":"resource not found: keypair"}
│

```

The used is key is available to the infra user as visible from `openstack keypair list`. Additionally a server is being deployed when using `openstack server create` with the same keypair id.  

Not sure if this is related to #578 

**To Reproduce**

Create a simple server resource

```
resource "stackit_server" "test" {
  project_id = "${var.project_id}"
  name       = "test-server"
  boot_volume = {
    size        = 30
    source_type = "image"
    source_id   = "3b113c81-18d2-4ddb-9876-7f0c95328788" #Rocky Linux 9 (2024-05-28)
  }
  availability_zone = "eu01-1"
  machine_type      = "g1.1"
  keypair_name      = "my_key"
  user_data    = file("${path.module}/cloud-init.yaml")
}

```
Execute `terraform apply`

**Expected behavior**

Server should be created.

**Environment:**

```
Terraform v1.9.4
on windows_amd64
+ provider registry.terraform.io/stackitcloud/stackit v0.33.2
```

**Comment by joaopalet** [comment]
Hi @alipievg,

Thanks for reporting the issue. 

We have no yet released the `stackit_key_pair` resource, so it's it not related to that. We will check it and will keep up updated here.

**Comment by joaopalet** [comment]
Hi @alipievg, 

This occurs because the OpenStack and STACKIT key pairs are disassociated.

So in order to use a key pair on a server created with the STACKIT provider, you would need to import the public key of the key pair on STACKIT. Currently you can do this via the Portal, but very soon we will release a new `stackit_key_pair` resource which will allow you to do exactly that.

**Comment by joaopalet** [comment]
Hi @alipievg,

The `stackit-key_pair` resource was released in [`v0.34.0`](https://registry.terraform.io/providers/stackitcloud/stackit/0.34.0).
